internal/gcsx/syncer.go

```diff
@@ -154,7 +154,7 @@ type syncer struct {
 func (os *syncer) SyncObject(
 	ctx context.Context,
 	srcObject *gcs.Object,
-	content TempFile) (o *gcs.Object, err error) {
+	content TempFile, localFileCache bool) (o *gcs.Object, err error) {
```

Comment:
Can we make the changes of syncer.go out of this PR? It isn't relevant as syncer is only used for the write path.

For the read path, the cache files won't be deleted.
For the write path, the cache files have to be deleted anyway.

So, I'm not sure why this has to be included in this PR.

Reply:
I have rolled the changes from #614 here so that we don't modify the syncer interface.  However, in order to preserve original functionality, I have added a check to destroy the temp file if local file cache is disabled.